synchronized port scanner.py

```diff
@@ -0,0 +1,51 @@
+"""
+Synchronized port scanner.
+"""
+
+from socket import socket, AF_INET, SOCK_STREAM
+import time
+
+
+def connect_socket(guest: socket, ip: str, port: int):
+    """
+    Connect given socket to given ip and port
+
+   :param guest: The socket to connect.
+   :param ip: The ip for the socket to connect.
+   :param port: The port for the socket to connect.
+    """
+    try:
+        guest.connect((ip, port))
+        print(f"port {port} found")
+    except ConnectionError:
+        print("connection failed!")
+
+
+def time_measure(func):
+    """
+    Measure time for execution
+
+    :param func: The function to be measured.
+    """
+    def wrapper(*args, **kw):
+        start_time = time.time()
+        func(*args, **kw)
+        end_time = time.time()
+        print(f"{end_time - start_time} seconds took!")
+        return func
+    return wrapper
```

Comment:
This function is used in both modules. You can export it to `utils.py` module or something so that you avoid code duplication.

Reply:
Similar duplicates can be added to the additional module and that way you can avoid all duplications.

Reply:
They aren't exactly similar. There is a small change for async method.